e2e/aks_model.go

```diff
@@ -19,6 +19,37 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/privatedns/armprivatedns"
 )
 
+// getLatestGAKubernetesVersion returns the latest GA Kubernetes version for the given location.
+func getLatestGAKubernetesVersion(location string) (string, error) {
+	versions, err := config.Azure.AKS.ListKubernetesVersions(context.Background(), location, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to list Kubernetes versions: %w", err)
+	}
+	if versions.Values == nil || len(versions.Values) == 0 {
+		return "", fmt.Errorf("no Kubernetes versions available")
+	}
+	for _, k8s_version := range versions.Values {
+		if k8s_version.Version == nil {
+			continue
+		}
+		if k8s_version.IsPreview == nil || !*k8s_version.IsPreview {
+			return *k8s_version.Version, nil
+		}
+	}
+	return "", fmt.Errorf("no GA Kubernetes version found")
+}
+
+// getLatestKubernetesVersionClusterModel returns a cluster model with the latest GA Kubernetes version.
+func getLatestKubernetesVersionClusterModel(name string) (*armcontainerservice.ManagedCluster, error) {
+	version, err := getLatestGAKubernetesVersion(config.Config.Location)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get latest GA Kubernetes version: %w", err)
+	}
+	model := getBaseClusterModel(name)
+	model.Properties.KubernetesVersion = to.Ptr(version)
```

Comment:
I believe it gets defaulted AKS RP anyways though I'd consider also setting `model.Properties.AgentPoolProfiles[0].OrchestratorVersion = to.Ptr(version)` as well

Reply:
no need, like you said, the AgentPool versionfollows the control-plane version :)